src/app/view/perspectives/chart/visualizer/plot-maker/bar-plot-maker.ts

```diff
@@ -142,39 +173,50 @@ export class BarPlotMaker extends PlotMaker {
   }
 
   public createLayout(): Partial<Layout> {
-    if (this.config.axes[ChartAxisType.Y2]) {
-      return {
-        barmode: 'group',
-        yaxis2: {
-          overlaying: 'y',
-          side: 'right',
-        },
-        legend: {
-          xanchor: 'left',
-          x: 1.1,
-        },
-      };
-    }
-    return {};
+    return {...this.yAxis1Layout(), ...this.yAxis2Layout(), ...this.otherLayout()};
+  }
+
+  private otherLayout(): Partial<Layout> {
+    return {
+      barmode: 'group',
+      legend: {
+        xanchor: 'left',
+        x: 1.1,
+      },
+    };
   }
 
   public currentType(): ChartType {
     return ChartType.Bar;
   }
 
-  public initDrag() {
-    // TODO
+  public getPointPosition(point: any, datum: any): {x: number; y: number} {
+    return {x: datum.x, y: point.clickedY};
   }
 
-  public destroyDrag() {
-    // TODO
+  public getTraceIndexForPoint(point: any): number {
+    const barsContainers = d3.selectAll('.barlayer .trace .points')[0];
+    const pointNode = d3.select(point).node().parentNode;
+
+    for (let i = 0; i < barsContainers.length; i++) {
+      const children = barsContainers[i].children;
+
+      if (Array.from(children).find(p => p === pointNode)) {
+        return i;
+      }
+    }
+
+    return 0;
   }
 
-  public dragEnabledChange() {
-    // TODO
+  public getPointNewY(point: any, datum: any, event: any): number {
+    const pointData: PointData = point.pointData;
+    const computedY = event.sourceEvent.pageY - pointData.offset.top + datum.ct[1];
+    const dy = computedY - pointData.clickedY;
+    return pointData.initialY + dy;
   }
 
-  public onRelayout() {
-    // TODO
+  public getPoints(): any {
+    return d3.selectAll('.barlayer .trace .points .point path');
```

Comment:
What about adding [@types/d3](https://www.npmjs.com/package/@types/d3) dependency to the project?